instrumentation/opentelemetry/readbackspan_test.go

```diff
@@ -0,0 +1,31 @@
+package opentelemetry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/hypertrace/goagent/sdk"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSetAndGetAttributeSuccess(t *testing.T) {
+	_, s, _ := StartReadbackSpan(context.Background(), "test_readbackspan", &sdk.SpanOptions{})
+	s.SetAttribute("test_key_1", true)
+	s.SetAttribute("test_key_2", int64(1))
+	s.SetAttribute("test_key_3", float64(1.2))
+	s.SetAttribute("test_key_4", "abc")
+	err := errors.New("xyz")
+	s.SetAttribute("test_key_4", err)
+	attributes := s.GetAttributes()
+	assert.Equal(t, attributes["test_key_1"].(bool), true)
+	assert.Equal(t, attributes["test_key_2"].(int64), int64(1))
+	assert.Equal(t, attributes["test_key_3"].(float64), 1.2)
+	assert.Equal(t, attributes["test_key_4"].(error), err)
+
+	assert.Equal(t, fmt.Sprintf("%v", attributes["test_key_1"].(bool)), "true")
```

Comment:
I am curious about these assertions, why exactly we do this given we have assertions on top?

Reply:
I duplicated it thinking it didn't hurt.  This is most likely due to my Go inexperience and just learning about interface{}'s .  From the consuming end of GetAttributes(), we always use a string to pass into libtraceable. 

I'll remove the second block of asserts.